framework/framework.go

```diff
@@ -99,25 +72,21 @@ func New(config Config) (*Framework, error) {
 		return nil, microerror.Maskf(invalidConfigError, "config.ResourceRouter must not be empty")
 	}
 
-	// Settings.
 	if config.BackOffFactory == nil {
-		return nil, microerror.Maskf(invalidConfigError, "config.BackOffFactory must not be empty")
+		config.BackOffFactory = DefaultBackOffFactory()
 	}
 
 	f := &Framework{
-		// Dependencies.
 		crd:            config.CRD,
 		crdClient:      config.CRDClient,
 		informer:       config.Informer,
 		logger:         config.Logger,
 		resourceRouter: config.ResourceRouter,
 
-		// Settings.
-		backOffFactory: config.BackOffFactory,
-
-		// Internals.
 		bootOnce: sync.Once{},
 		mutex:    sync.Mutex{},
+
+		backOffFactory: config.BackOffFactory,
```

Comment:
The order is dependencies, internals, settings. Hashtag alphabet.

Reply:
Alphabet collides with logic here. Internals are constructed from dependencies and settings and I think it's easier to maintain it like that.

In New you validate dependencies and settings first, so it's easier to not miss any if they aren't intertwined with settings

Reply:
This is also documented here: https://github.com/giantswarm/fmt/pull/4/files#diff-09196e856cd1c9274e7747be39ebfbdbR200

Reply:
It was always dependencies, internals, settings from the very beginning. It looks like I did not pay that much attention to the formatting rules we documented or I missed that detail. 

> Internals are constructed from dependencies and settings

This does only represent some cases. As you see here, this is not the case.

Reply:
Yes, but in general it's good to see what is copied directly from Config struct. That's practical thing.

Reply:
I think you see what is copied. The order does not matter for this to recognize. Lets talk next week so we can agree on something for the long run.